stm32-modules/heater-shaker/simulator/motor_thread.cpp

```diff
@@ -59,6 +67,11 @@ auto run(std::stop_token st, std::shared_ptr<TaskControlBlock> tcb) -> void {
         } catch (const SimMotorTask::Queue::StopDuringMsgWait& sdmw) {
             return;
         }
+        if (tcb->task.get_state() ==
+            motor_task::State::HOMING_COASTING_TO_STOP) {
```

Comment:
I'm a bit confused, is coasting to stop bad? Why are we trying to send an error, or am I misunderstanding the code?

Reply:
Well, the error message is kind of internal to the motor task. This is emulating the motor  driver saying "i had an overcurrent event", aka "i stalled because the solenoid is engaged". the message is sent  to the motor task, and therefore exercises the code on line 179 of motor_task.hpp (sadly tough to interlink) above. This is the way we tell the motor task that homing is done, on th simulation side.